core/services/runs.go

```diff
@@ -370,6 +370,11 @@ func validateOnMainChain(jr *models.JobRun, taskRun *models.TaskRun, store *stor
 	return nil
 }
 
+func invalidRequest(request models.RunRequest, receipt *models.TxReceipt) bool {
+	return receipt.Unconfirmed() ||
+		(request.BlockHash != nil && *request.BlockHash != receipt.BlockHash)
```

Comment:
How does `request.BlockHash = nil` happen?

Reply:
Old entries that were `in progress` as we roll out an update and bring back up the instance?

Reply:
Like @dimroc said, instead of backfilling we have this check. Could probably get rid of it eventually, but it will smooth rollout now.

Reply:
Thanks, guys.